tests/test_transforms/test_decomposition/test_deseasonal_transform.py

```diff
@@ -0,0 +1,184 @@
+    pd.util.testing.assert_frame_equal(df_inverse_transformed, df)
+
+
+@pytest.mark.parametrize("model", ["additive", "multiplicative"])
+@pytest.mark.parametrize("ts_name", ["ts_seasonal"])
+def test_inverse_transform_multi_segments(ts_name, model, request):
+    """Test that transform + inverse_transform don't change tsdataset."""
+    ts = request.getfixturevalue(ts_name)
+    transform = DeseasonalityTransform(in_column="target", period=7, model=model)
+    df = ts.to_pandas(flatten=True)
+    transform.fit_transform(ts)
+    transform.inverse_transform(ts)
+    df_inverse_transformed = ts.to_pandas(flatten=True)
+    pd.util.testing.assert_frame_equal(df_inverse_transformed, df)
+
+
+@pytest.mark.parametrize("model_decompose", ["additive", "multiplicative"])
+def test_forecast(ts_seasonal, model_decompose):
+    """Test that transform works correctly in forecast."""
+    transform = DeseasonalityTransform(in_column="target", period=7, model=model_decompose)
+    ts_train, ts_test = ts_seasonal.train_test_split(
+        ts_seasonal.index[0],
+        ts_seasonal.index[-4],
+        ts_seasonal.index[-3],
+        ts_seasonal.index[-1],
+    )
+    transform.fit_transform(ts_train)
+    model = NaiveModel()
+    model.fit(ts_train)
+    ts_future = ts_train.make_future(future_steps=3, transforms=[transform], tail_steps=model.context_size)
+    ts_forecast = model.forecast(ts_future, prediction_size=3)
+    ts_forecast.inverse_transform([transform])
+    for segment in ts_forecast.segments:
+        np.testing.assert_allclose(ts_forecast[:, segment, "target"], ts_test[:, segment, "target"], atol=0.1)
+
+
+def test_transform_raise_error_if_not_fitted(df_seasonal_one_segment):
+    """Test that transform for one segment raise error when calling transform without being fit."""
+    transform = _OneSegmentDeseasonalityTransform(in_column="target", period=7, model="additive")
+    with pytest.raises(ValueError, match="Transform is not fitted!"):
+        _ = transform.transform(df=df_seasonal_one_segment)
+
+
+def test_inverse_transform_raise_error_if_not_fitted(df_seasonal_one_segment):
+    """Test that transform for one segment raise error when calling inverse_transform without being fit."""
+    transform = _OneSegmentDeseasonalityTransform(in_column="target", period=7, model="additive")
+    with pytest.raises(ValueError, match="Transform is not fitted!"):
+        _ = transform.inverse_transform(df=df_seasonal_one_segment)
+
+
+@pytest.mark.parametrize("model_decompose", ["additive", "multiplicative"])
+def test_fit_transform_with_nans_in_tails(ts_seasonal_nan_tails, model_decompose):
+    transform = DeseasonalityTransform(in_column="target", period=7, model=model_decompose)
+    transform.fit_transform(ts=ts_seasonal_nan_tails)
+    np.testing.assert_allclose(ts_seasonal_nan_tails[:, :, "target"].dropna(), 10, atol=0.25)
+
+
+def test_fit_transform_with_nans_in_head_or_middle_raise_error(ts_with_nans):
```

Comment:
What is wrong with NaNs in the beginning? Because we suppose that it is ok to have NaNs in the beginning.